vscode/snippets/ei-snippets/src/main/java/org/wso2/integration/ballerina/util/SnippetsBlock.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.wso2.integration.ballerina.util;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.ballerinalang.langserver.compiler.LSContext;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionItemKind;
+import org.eclipse.lsp4j.TextEdit;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SnippetsBlock {
+     private String label = "";
+    private String detail = "";
+    private String snippet;
+    private SnippetType snippetType;
+    private final Pair<String, String>[] imports;
+
+    public SnippetsBlock(String snippet, SnippetType snippetType) {
+        this.snippet = snippet;
+        this.snippetType = snippetType;
+        this.imports = null;
+    }
+
+    public SnippetsBlock(String label, String snippet, String detail, SnippetType snippetType) {
+        this.label = label;
+        this.snippet = snippet;
+        this.detail = detail;
+        this.snippetType = snippetType;
+        this.imports = null;
+    }
+
+    public SnippetsBlock(String label, String snippet, String detail, SnippetType snippetType,
+                         Pair<String, String>... importsByOrgAndAlias) {
+        this.label = label;
+        this.snippet = snippet;
+        this.detail = detail;
+        this.snippetType = snippetType;
+        this.imports = importsByOrgAndAlias;
+    }
+
+    /**
+     * Create a given completionItem's insert text.
+     *
+     * @param ctx   LS Context
```

Comment:
what is `LSContext`? Better to explain little be more.

Reply:
Fixed